packages/melos/lib/src/common/persistent_shell.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import '../logging.dart';
+import 'platform.dart';
+import 'utils.dart';
+
+class PersistentShell {
+  PersistentShell({
+    required this.logger,
+    this.workingDirectory,
+  });
+
+  final _isWindows = currentPlatform.isWindows;
+  final MelosLogger logger;
+  final String? workingDirectory;
+  late final Process _process;
+  Completer<void>? _commandCompleter;
+
+  Future<void> startShell() async {
+    final executable = _isWindows ? 'cmd.exe' : '/bin/sh';
+
+    _process = await Process.start(
+      executable,
+      _isWindows ? ['/C'] : [],
+      workingDirectory: workingDirectory,
+    );
+
+    _process.stdout.listen(
+      (event) {
+        final output = utf8.decode(event, allowMalformed: true);
+        _handleCommandCompletion(output);
+        logger.logWithoutNewLine(output);
```

Comment:
I think we can omit printing the labels here and just print the actual command output

Reply:
by labels you mean 'success', 'running' or 'failed'?

Reply:
Exactly, and then at then end of all steps we can print whether it was a success or failed instead